File: ios/Plugin/openSubscriptionSettingsPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(openSubscriptionSettingsPlugin, "openSubscriptionSettings",
           CAP_PLUGIN_METHOD(triggerOpenSubscriptions, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(triggerOpenSettings, CAPPluginReturnPromise);
)

File: ios/Plugin/openSubscriptionSettingsPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(openSubscriptionSettingsPlugin)
public class openSubscriptionSettingsPlugin: CAPPlugin {
    @objc func triggerOpenSettings(_ call: CAPPluginCall) {
        guard let url = URL(string: UIApplication.openSettingsURLString) else { return }
        UIApplication.shared.open(url, options: [:])
    }

    @objc func triggerOpenSubscriptions(_ call: CAPPluginCall) {
        guard let url = URL(string: "itms://apps.apple.com/account/subscriptions") else { return }
        UIApplication.shared.open(url, options: [:])
    }
}
